src/main/scala/io/qbeast/spark/delta/DefaultFileIndex.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2021 Qbeast Analytics, S.L.
+ */
+package io.qbeast.spark.delta
+
+import io.qbeast.spark.index.query.QueryFiltersUtils
+import org.apache.hadoop.fs.Path
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.delta.files.TahoeLogFileIndex
+import org.apache.spark.sql.delta.DeltaLog
+import org.apache.spark.sql.execution.datasources.FileIndex
+import org.apache.spark.sql.execution.datasources.PartitionDirectory
+import org.apache.spark.sql.execution.SQLExecution
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.SparkSession
+
+/**
+ * Default implementation of the FileIndex.
+ *
+ * @param target
+ *   the target file index implemented by Delta
+ */
+class DefaultFileIndex private (target: TahoeLogFileIndex)
+    extends FileIndex
+    with QueryFiltersUtils
+    with Logging
+    with Serializable {
+
+  override def rootPaths: Seq[Path] = target.rootPaths
+
+  override def listFiles(
+      partitionFilters: Seq[Expression],
+      dataFilters: Seq[Expression]): Seq[PartitionDirectory] = {
+    logFilters(partitionFilters, dataFilters)
+    val strategy = if (haveQbeastWeightExpression(dataFilters)) {
+      SamplingListFilesStrategy
+    } else {
+      DefaultListFilesStrategy
+    }
+    strategy.listFiles(target, partitionFilters, dataFilters)
+  }
+
+  private def logFilters(
```

Comment:
Instead of having a separate method in the class, wouldn't be better to implement an object for Logging where we can also add other types? (Like logging for writing, optimizing....)

I know it might be out of the scope of this PR, but if it's necessary we can open an issue and another PR for that.

Reply:
IMHO it is too early, do you know other pieces of code which need such an utility object? Anyway feel free to open a separate issue.

Reply:
I will open an issue with some other processes that might need Logging information, and which type of information could be good to show to the developer. Thanks.